### PR TITLE
TPD-936 new special link giving pages, TPD-812 Make Processing Error Codes More Clear

### PR DESCRIPTION
**Pull request by jason-rice**
[touchpointsoftware.atlassian.net/browse/TPD-936](https://touchpointsoftware.atlassian.net/browse/TPD-936)
[touchpointsoftware.atlassian.net/browse/TPD-812](https://touchpointsoftware.atlassian.net/browse/TPD-812)

### CODE REVIEW

File: CmsWeb/Areas/Giving/Controllers/GivingManagementController.cs [hunk 1]
```diff
@@ -22,6 +24,31 @@ public GivingManagementController(IRequestManager requestManager) : base(request
         [Route("~/Giving")]
         public ActionResult Index()
         {
+            //GivingPaymentViewModel viewModel = new GivingPaymentViewModel()
```
Comment (timdavila) [inline CmsWeb/Areas/Giving/Controllers/GivingManagementController.cs:27]: please don't leave commented code like this unless it has a purpose

File: CmsWeb/Areas/Giving/Models/GivingPageModel.cs [hunk 2]
```diff
@@ -206,6 +206,28 @@ public GivingPageItem UpdateGivingDefaultPage(bool value, int PageId)
...
+                foreach (var item in givingPage.GivingPageFunds)
+                {
+                    var fund = new ContributionFund()
+                    {
+                        FundId = item.ContributionFund.FundId,
+                        FundName = item.ContributionFund.FundName
+                    };
+                    fundsList.Add(fund);
+                }
+                fundsList.OrderBy(f => f.FundName);
```
Comment (timdavila) [inline CmsWeb/Areas/Giving/Models/GivingPageModel.cs:227]: funds have a sort order column defined by the user that we would usually order by. i'm not sure if it's relevant here, just wanted to be sure you were aware.

File: CmsWeb/Areas/Giving/Models/GivingPaymentModel.cs [hunk 3]
```diff
@@ -317,9 +320,16 @@ public Message CreateSchedule(GivingPaymentViewModel viewModel)
                     return Message.createErrorReturn("Contribution fund not found.", Message.API_ERROR_SCHEDULED_GIFT_FUND_ID_NOT_FOUND);
             }
 
+            // this is for testing purposes
```
Comment (timdavila) [inline CmsWeb/Areas/Giving/Models/GivingPaymentModel.cs:323]: good use of comment here

File: CmsWeb/Areas/Giving/Controllers/GivingManagementController.cs (hunk 1 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Controllers/GivingManagementController.cs:27]: yeah, I'm glad you noticed this. Forgot to delete this, it was just for testing

File: CmsWeb/Areas/Giving/Models/GivingPaymentModel.cs (hunk 3 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPaymentModel.cs:323]: thanks

File: CmsWeb/Areas/Giving/Models/GivingPageModel.cs (hunk 2 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPageModel.cs:227]: yeah this method is for displaying the list of funds when creating a special link for sms and email. I wanted to give the user a way to find the fund they're looking for, without this orderby, the list would be random and difficult for the user to find the fund they want

**Pull request merged**